### DISCUSSION TITLE
How does pnpm's repo as a monorepo handle semantic release automation? Does it use semantic-release?

### DISCUSSION THREAD

**Discussion by 205g0** [pnpm/pnpm#3426] [Q&A]
Pretty much the title...

**Comment by zkochan** [comment]
We use changesets: https://pnpm.io/using-changesets

**Reply by 205g0** [reply]
Thanks for the hint. I played around a bit with changesets and how is your workflow (just for create the release docs, not the actual npm publishes)?

1. Development happens in branches which get pushed
2. Before they `git merge`d you run a `pnpx changeset version`
3. Once merged on GH the changset action kicks in

But do you still trigger the changelog/release manually or is it done automatically?

**Reply by zkochan** [reply]
I do not run `pnpx changeset version` before merge. All the changeset files are merged to `main`. When I want to make a release, I run `changeset version && pnpm install && pnpm publish -r` on main branch.

**Comment by reuzel** [comment]
I'm using [multi-semantic-release](https://www.npmjs.com/package/multi-semantic-release) in my repository. All packages in my monorepo have version 0.0.0-development in the code base. Multi-semantic-release, based on commit history will assign the right version numbers and dependencies before publishing to my package repo. It does this smart: a package version will also be bumped if on of the dependencies updates.
